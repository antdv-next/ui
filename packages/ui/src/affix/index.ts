import type { App } from 'vue'
import Affix from './affix.vue'

export type { AffixProps } from './define'
Affix.install = (app: App) => {
  app.component('AAffix', Affix)
}
export {
  Affix,
}
