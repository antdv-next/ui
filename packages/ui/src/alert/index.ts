import type { App } from 'vue'
import Alert from './alert.vue'

export type { AlertProps } from './define'

Alert.install = (app: App) => {
  app.component('AAlert', Alert)
}

export {
  Alert,
}

export default Alert
