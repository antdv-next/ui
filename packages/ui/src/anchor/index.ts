import type { App } from 'vue'
import AnchorLink from './anchor-link.vue'
import Anchor from './anchor.vue'

export type { AnchorLinkBaseProps, AnchorProps } from './define'

Anchor.install = (app: App) => {
  app.component('AAnchor', Anchor)
  app.component('AAnchorLink', AnchorLink)
}

AnchorLink.install = (app: App) => {
  app.component('AAnchorLink', AnchorLink)
}

export {
  Anchor,
  AnchorLink,
}
export default Anchor
