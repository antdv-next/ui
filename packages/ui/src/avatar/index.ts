import type { App, Plugin } from 'vue'
import Avatar from './avatar.vue'
import Group from './group.vue'

export type { AvatarGroupProps, AvatarProps } from './define'

Avatar.install = function (app: App) {
  app.component(Avatar.name!, Avatar)
  app.component(Group.name!, Group)
  return app
}

Avatar.Group = Group

export { Avatar, Group as AvatarGroup }

export default Avatar as typeof Avatar
  & Plugin & {
    readonly Group: typeof Group
  }
