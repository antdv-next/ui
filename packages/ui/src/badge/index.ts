import type { App, Plugin } from 'vue'
import Badge from './badge.vue'
import Ribbon from './ribbon.vue'

export type { BadgeProps } from './define'

Badge.install = function (app: App) {
  app.component(Badge.name!, Badge)
  app.component(Ribbon.name!, Ribbon)
  return app
}

export { Badge, Ribbon as BadgeRibbon }

export default Badge as typeof Badge
  & Plugin & {
    readonly Ribbon: typeof Ribbon
  }
