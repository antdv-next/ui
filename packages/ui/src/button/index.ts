import type { App } from 'vue'
import Button from './button.vue'

export type { ButtonProps } from './define'
Button.install = (app: App) => {
  app.component('AButton', Button)
}

Button.__ANT_BUTTON = true

export {
  Button,
}
