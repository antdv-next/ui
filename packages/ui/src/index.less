@import './style/index.css';
@import './wave/style/index.less';
@import './affix/style/index.less';
@import './button/style/index.less';
@import './alert/style/index.less';
@import './divider/style/index.less';
@import './space/style/index.less';
@import './anchor/style/index.less';
@import './tooltip/style/index.less';
@import './popover/style/index.less';
@import './spin/style/index.less';
@import './qrcode/style/index.less';
@import './flex/style/index.less';
@import './popconfirm/style/index.less';
