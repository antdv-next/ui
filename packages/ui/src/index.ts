// 主题相关导出
export * from './theme/cssvar/genCssvar'
export * from './theme/cssvar/genStyle'
export * from './theme/cssvar/parseStyle'

// 主题接口类型
export type * from './theme/interface/alias'
export type * from './theme/interface/components'
export type * from './theme/interface/maps'
export type * from './theme/interface/presetColors'
export type * from './theme/interface/seeds'

// 主题算法和配置
export * from './theme/themes/compact'
export * from './theme/themes/dark'
export * from './theme/themes/default'
export * from './theme/themes/seed'
