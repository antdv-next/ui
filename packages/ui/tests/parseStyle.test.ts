import type { CSSObject, ParseConfig } from '../src'
import { describe, expect, it } from 'vitest'
import { parseStyle, parseStyleInterpolation, parseStyleToLess } from '../src'

describe('parseStyle', () => {
  it('should parse basic CSS properties', () => {
    const styles: CSSObject = {
      color: 'red',
      fontSize: 16,
      padding: '10px',
    }

    const result = parseStyleToLess(styles)
    expect(result).toContain('color:red;')
    expect(result).toContain('font-size:16px;')
    expect(result).toContain('padding:10px;')
  })

  it('should handle nested selectors', () => {
    const styles: CSSObject = {
      'color': 'blue',
      '&:hover': {
        color: 'red',
      },
      '.child': {
        marginTop: 8,
      },
    }

    const result = parseStyleToLess(styles)
    expect(result).toContain('color:blue;')
    expect(result).toContain('&:hover{color:red;}')
    expect(result).toContain('.child{margin-top:8px;}')
  })

  it('should handle media queries', () => {
    const styles: CSSObject = {
      'fontSize': 16,
      '@media (max-width: 768px)': {
        fontSize: 14,
      },
    }

    const result = parseStyleToLess(styles)
    expect(result).toContain('font-size:16px;')
    expect(result).toContain('@media (max-width: 768px){font-size:14px;}')
  })

  it('should inject hash ID correctly', () => {
    const styles: CSSObject = {
      'color': 'red',
      '&:hover': {
        color: 'blue',
      },
    }

    const config: ParseConfig = {
      hashId: 'test-123',
    }

    const result = parseStyleToLess(styles, config)
    expect(result).toContain('color:red;')
    expect(result).toContain('.test-123:hover{color:blue;}')
  })

  it('should handle complex nested structures', () => {
    const styles: CSSObject = {
      'position': 'relative',
      '.header': {
        'padding': 16,
        '&.active': {
          fontWeight: 'bold',
        },
      },
    }

    const result = parseStyleToLess(styles)
    expect(result).toContain('position:relative;')
    expect(result).toContain('.header{padding:16px;&.active{font-weight:bold;}}')
  })

  it('should add px units to numeric values when needed', () => {
    const styles: CSSObject = {
      width: 100,
      height: 200,
      zIndex: 1000, // should not add px
      lineHeight: 1.5, // should not add px
    }

    const result = parseStyleToLess(styles)
    expect(result).toContain('width:100px;')
    expect(result).toContain('height:200px;')
    expect(result).toContain('z-index:1000;')
    expect(result).toContain('line-height:1.5;')
  })

  it('should handle camelCase to kebab-case conversion', () => {
    const styles: CSSObject = {
      backgroundColor: 'white',
      borderRadius: 4,
      marginTop: 8,
    }

    const result = parseStyleToLess(styles)
    expect(result).toContain('background-color:white;')
    expect(result).toContain('border-radius:4px;')
    expect(result).toContain('margin-top:8px;')
  })

  it('should work with original interface', () => {
    const styles = {
      color: 'red',
      fontSize: 16,
    }

    const mockToken = {} as any
    const result = parseStyle(styles, mockToken)
    expect(result).toContain('color:red;')
    expect(result).toContain('font-size:16px;')
  })

  it('should handle empty styles', () => {
    const styles: CSSObject = {}
    const result = parseStyleToLess(styles)
    expect(result).toBe('')
  })

  it('should handle layer configuration', () => {
    const styles: CSSObject = {
      color: 'red',
    }

    const config: ParseConfig = {
      layer: {
        name: 'components',
        dependencies: ['base', 'theme'],
      },
    }

    const [styleStr, effectStyle] = parseStyleInterpolation(styles, config)
    expect(styleStr).toContain('@layer components')
    expect(effectStyle['@layer components']).toContain('@layer base, components;')
    expect(effectStyle['@layer components']).toContain('@layer theme, components;')
  })

  it('should handle string interpolation', () => {
    const styles = [
      'color: red;',
      {
        fontSize: 16,
      },
      'margin: 0;',
    ]

    const [result] = parseStyleInterpolation(styles)
    expect(result).toContain('color: red;')
    expect(result).toContain('font-size:16px;')
    expect(result).toContain('margin: 0;')
  })

  it('should handle deep nesting correctly', () => {
    const styles: CSSObject = {
      '.container': {
        '.header': {
          '.title': {
            'fontSize': 18,
            '&.large': {
              fontSize: 24,
            },
          },
        },
      },
    }

    const result = parseStyleToLess(styles)
    expect(result).toContain('.container{.header{.title{font-size:18px;&.large{font-size:24px;}}}}')
  })

  it('should handle hash priority', () => {
    const styles: CSSObject = {
      'color': 'red',
      '.child': {
        color: 'blue',
      },
    }

    const highPriorityConfig: ParseConfig = {
      hashId: 'test-hash',
      hashPriority: 'high',
    }

    const lowPriorityConfig: ParseConfig = {
      hashId: 'test-hash',
      hashPriority: 'low',
    }

    const highResult = parseStyleToLess(styles, highPriorityConfig)
    const lowResult = parseStyleToLess(styles, lowPriorityConfig)

    // 高优先级会把 hash 放在前面
    expect(highResult).toContain('.test-hash .child')
    // 低优先级会把 hash 放在后面
    expect(lowResult).toContain('.child.test-hash')
  })

  it('should not add px units to z-index CSS variables', () => {
    const styles: CSSObject = {
      // 传统 z-index 属性
      'zIndex': 1000,
      // CSS 变量形式的 z-index
      '--ant-z-index': 1001,
      // 其他数值属性应该添加 px
      'width': 100,
      'height': 200,
      // CSS 变量但非 z-index 相关
      '--ant-color-primary': 16,
    }
    const result = parseStyleToLess(styles)

    // z-index 相关属性不应该有 px
    expect(result).toContain('z-index:1000;')
    expect(result).toContain('--ant-z-index:1001;')

    // 其他数值属性应该有 px
    expect(result).toContain('width:100px;')
    expect(result).toContain('height:200px;')
    expect(result).toContain('--ant-color-primary:16px;')
  })

  it('should handle custom prefix for z-index CSS variables', () => {
    const styles: CSSObject = {
      '--ant-custom-z-index': 1002,
      '--ant-custom-z-index-mask': 1003,
      '--ant-custom-z-index-overlay': 1004,
    }

    const config: ParseConfig = {
      prefixCls: 'ant-custom',
    }

    const result = parseStyleToLess(styles, config)

    expect(result).toContain('--ant-custom-z-index:1002;')
    expect(result).toContain('--ant-custom-z-index-mask:1003;')
    expect(result).toContain('--ant-custom-z-index-overlay:1004;')
  })

  it('should format CSS with proper indentation and line breaks', () => {
    const styles: CSSObject = {
      'color': 'red',
      'fontSize': 16,
      'padding': '10px',
      '&:hover': {
        color: 'blue',
        fontSize: 18,
      },
      '.child': {
        marginTop: 8,
        backgroundColor: 'white',
      },
    }

    const formattedResult = parseStyleToLess(styles, { formatStyle: true })
    const unformattedResult = parseStyleToLess(styles, { formatStyle: false })

    // 格式化的结果应该包含换行和缩进
    expect(formattedResult).toContain('  color: red;\n')
    expect(formattedResult).toContain('  font-size: 16px;\n')
    expect(formattedResult).toContain('&:hover {\n')
    expect(formattedResult).toContain('  color: blue;\n')
    expect(formattedResult).toContain('}\n')

    // 未格式化的结果应该是一行
    expect(unformattedResult).toContain('color:red;')
    expect(unformattedResult).toContain('font-size:16px;')
    expect(unformattedResult).not.toContain('\n  ')
  })

  it('should handle nested formatting correctly', () => {
    const styles: CSSObject = {
      '@media (max-width: 768px)': {
        'fontSize': 14,
        '.container': {
          'padding': 16,
          '&:hover': {
            opacity: 0.8,
          },
        },
      },
    }

    const result = parseStyleToLess(styles, { formatStyle: true })

    // 检查嵌套缩进
    expect(result).toContain('@media (max-width: 768px) {\n')
    expect(result).toContain('  font-size: 14px;\n')
    expect(result).toContain('.container {\n')
    expect(result).toContain('    padding: 16px;\n')
    expect(result).toContain('&:hover {\n')
    expect(result).toContain('      opacity: 0.8;\n')
  })
})
