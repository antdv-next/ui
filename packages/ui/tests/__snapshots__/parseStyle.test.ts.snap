// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parseStyle > should handle complex nested structures 1`] = `
"position: relative;
.header {
  padding: 16px;
  &.active {
    font-weight: bold;
  }
}
"
`;

exports[`parseStyle > should handle deep nesting correctly 1`] = `
".container {
  .header {
    .title {
      font-size: 18px;
      &.large {
        font-size: 24px;
      }
    }
  }
}
"
`;

exports[`parseStyle > should handle media queries 1`] = `
"font-size: 16px;
@media (max-width: 768px) {
  font-size: 14px;
}
"
`;

exports[`parseStyle > should handle nested selectors 1`] = `
"color: blue;
&:hover {
  color: red;
}
.child {
  margin-top: 8px;
}
"
`;

exports[`parseStyle > should inject hash ID correctly 1`] = `
"color: red;
.test-123:hover {
  color: blue;
}
"
`;
